### ISSUE TITLE
using same (page) alias on two language tenants doesn't work

### ISSUE THREAD

**Issue by Knorrke** [serlo-org/athene2#714]
When the `Alias/Listener/PageControllerListener` calls the `createAlias` method of the manager, this call here checks for an existing alias:
https://github.com/serlo-org/athene2/blob/4cd7c5ed82c357daaae49c5e806af3ff4382128c/src/module/Alias/src/Alias/AliasManager.php#L117

This however doesn't use the instance id of the alias that should be created. This is the relevant code:
https://github.com/serlo-org/athene2/blob/4cd7c5ed82c357daaae49c5e806af3ff4382128c/src/module/Alias/src/Alias/AliasManager.php#L280-L309

Probably this can be fixed by passing the instance id, too and using it as criteria for the alias.

As a sidenote: Can't we use the storage here, too? Or better: Why do we use `findAliases` instead of `findSourceByAlias`?

**Comment by Knorrke** [comment]
Also... shouldn't `findSourceByAlias` use the instance too, to resolve the alias??
https://github.com/serlo-org/athene2/blob/4cd7c5ed82c357daaae49c5e806af3ff4382128c/src/module/Alias/src/Alias/AliasManager.php#L233-L235
I'm confused now...

**Comment by arekkas** [comment]
> As a sidenote: Can't we use the storage here, too? Or better: Why do we use findAliases instead of findSourceByAlias?

Because caching.

> Also... shouldn't findSourceByAlias use the instance too, to resolve the alias?

Yes, it should.

I think there was a reason for not having the instance here. I'll try to remember. Obviously it's non-sensical that this breaks duplicate aliases.

**Comment by arekkas** [comment]
So it seems like `instance` is being set and also used. I think it's simply a "whoops forgot to add that to criteria" thing on my part, so a good old bug. I suggest to try to recreate the bug locally, then add the instance to the parts where it's missing, and see if the affected pages start working then. You should also test this with a hot and cold cache test (for hot just reload a bunch of times, for cold restart the docker image). You should also test if urls that are not shared across instances work after the patch.

**Comment by Knorrke** [comment]
So, I was able to recreate the bug locally that aliases weren't created per instance but globally (the actual bug of this issue). I fixed that in the PR #738 and tested it as you explained. I also did a manual white-box test covering all branches of the alias manager, so this should be good to go

As for the `criteria` in `findSourceByAlias`: I have no idea why, but it actually did resolve the alias correctly no matter if I added the instance to criteria or not. At least I couldn't create any failing case whith two pages on different tenants having the same alias. I have no idea, why that worked, I also did disable caching. I changed that now anyway since it seemed to be more logical to have it there.
